sky/backends/cloud_vm_ray_backend.py

```diff
@@ -2327,6 +2374,32 @@ def is_provided_ips_valid(ips: Optional[List[Optional[str]]]) -> bool:
             internal_external_ips[1:], key=lambda x: x[1])
         self.stable_internal_external_ips = stable_internal_external_ips
 
+    @functools.lru_cache()
+    @timeline.event
+    def get_command_runners(
+            self,
+            avoid_ssh_control: bool = False
+    ) -> List[command_runner.CommandRunner]:
```

Comment:
Seems like a lot of usage of this function is only requiring the head runners. could we add an argument here and only construct and return the head runner?

Reply:
I am leaning toward not adding such argument as that will complicate the function interface for argument and return type.